tests/test_api_client.py

```diff
@@ -0,0 +1,59 @@
+from typing import Any, Iterator
+from unittest.mock import MagicMock
+
+import pytest
+import requests
+
+from bigc.api_client import BigCommerceRequestClient
+from bigc.exceptions import BigCommerceNetworkError
+
+
+class DummyBigCommerceRequestClient(BigCommerceRequestClient):
+    def get_many(
+        self,
+        path: str,
+        *,
+        page_size: int | None = None,
+        params: dict[str, Any] | None = None,
+        timeout: float | None = None,
+    ) -> Iterator[Any]:
+        raise NotImplementedError
+
+    def _prepare_url(self, path: str) -> str:
+        return f'https://api.example.com/stores/{self.store_hash}/test/{path.lstrip("/")}'
+
+
+@pytest.fixture
+def dummy_request_client() -> DummyBigCommerceRequestClient:
+    return DummyBigCommerceRequestClient('store_hash', 'access_token')
+
+
+@pytest.fixture
+def request_mock(monkeypatch):
+    mock_response = MagicMock(ok=True)
+    mock_response.text = '{"test": "response"}'
+    mock_response.json.return_value = {"test": "response"}
+
+    monkeypatch.setattr(
+        'bigc.api_client.requests.request',
+        mock := MagicMock(return_value=mock_response),
+    )
+
+    return mock
```

Comment:
Can we do some autospeccing here to improve our use of mocks?

Reply:
<img width="621" alt="image" src="https://github.com/user-attachments/assets/ad452809-0ca1-4fdd-8a21-09fef2d88837" />

Is this what you wanted?

Reply:
https://github.com/MedShift/bigc/pull/55/commits/c32b2f8d3a13bf48fe0d09169839e95162cb94f1